Mathlib/Data/Polynomial/Degree/Definitions.lean

```diff
@@ -1306,6 +1312,21 @@ theorem degree_sub_le_of_le {a b : WithBot ℕ} (hp : degree p ≤ a) (hq : degr
     degree (p - q) ≤ max a b :=
 (p.degree_sub_le q).trans <| max_le_max ‹_› ‹_›
 
+theorem leadingCoeff_sub_of_degree_lt [Ring R] (p q : R[X])
+    (h : Polynomial.degree q < Polynomial.degree p) : (p - q).leadingCoeff = p.leadingCoeff := by
+  rw [sub_eq_add_neg]
+  apply leadingCoeff_add_of_degree_lt_left
+  rw [degree_neg]
+  exact h
+
+theorem leadingCoeff_sub_of_degree_eq [Ring R] (p q : R[X]) (h : degree p = degree q)
+    (hlc : leadingCoeff p - leadingCoeff q ≠ 0) :
+    leadingCoeff (p - q) = leadingCoeff p - leadingCoeff q := by
+  rw [sub_eq_add_neg, leadingCoeff_add_of_degree_eq, leadingCoeff_neg, sub_eq_add_neg]
+  · rw [h, degree_neg]
+  · rw [leadingCoeff_neg, <-sub_eq_add_neg]
+    exact hlc
```

Comment:
Spawning subgoals by rewriting with missing hypotheses makes proofs harder to follow. It's better to avoid this.
```suggestion
theorem leadingCoeff_sub_of_degree_eq (h : degree p = degree q)
    (hlc : leadingCoeff p ≠ leadingCoeff q) :
    leadingCoeff (p - q) = leadingCoeff p - leadingCoeff q := by
  replace h : degree p = degree (-q) := by rwa [q.degree_neg]
  replace hlc : leadingCoeff p + leadingCoeff (-q) ≠ 0 := by
    rwa [← sub_ne_zero, sub_eq_add_neg, ← q.leadingCoeff_neg] at hlc
  rw [sub_eq_add_neg, leadingCoeff_add_of_degree_eq h hlc, leadingCoeff_neg, sub_eq_add_neg] 
```